doc/developer/design/20231110_aws_connections.md

```diff
@@ -107,16 +112,58 @@ When creating the connection, users should specify `WITH (VALIDATE = false)` in
 This is because we won't be able to validate the connection unless the trust policy has
 been added on the user's end.
 
+#### `mz_aws_connections` table
+
+We'll add a new `mz_aws_connections` table with the following definition:
+
+```sql
+CREATE TABLE mz_internal.mz_aws_connections (
+    -- The ID of the connection in `mz_connections`.
+    id text NOT NULL,
+    -- The value of the `ENDPOINT` option, if specified. `NULL` otherwise.
+    endpoint text,
+    -- The value of the `REGION` option, if specified. `NULL` otherwise.
+    region text,
+    -- The value of the `AWS ACCESS KEY ID` option, if specified as a string.
+    -- `NULL` otherwise.
+    access_key_id text,
+    -- The value of the `AWS ACCESS KEY ID` option, if specified as a secret.
+    -- `NULL` otherwise.
+    access_key_id_secret_id text,
```

Comment:
@benesch Ran into this while implementation: If access_key_id is a secret, then this would refer to a resolved item. We should show the fully resolved name right instead of what the user had provided?

Reply:
Ah, sorry, this should be the ID of the secret, referencing the row in the `mz_secrets` table! Then users can join with `mz_schemas` and `mz_databases` to get the full name. (The type of the column has to be `text` because the IDs are formatted like `u23` and `s34`; they're not simple numbers.) And apologies, it looks like I didn't get the doc comment on that field quite right.

Reply:
Ah thanks for clarifying. No worries, I will update the wording in the implementation PR.

Reply:
Updated here https://github.com/MaterializeInc/materialize/pull/23282/files#diff-67c46a1362121b3b9802f9fe608bd0cc9876f37c15c0e3559d1302b2c8f54a76R130

Reply:
Thank you!